### ISSUE TITLE
[Bug]: Unable to delete LDAP-user (remnants)

### ISSUE THREAD

**Issue by OneTwoBarbecue** [nextcloud/server#37632]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When trying to delete users shown in the remnants list with
sudo -u www-data php occ user:delete [nextcloudname]
the following errors are shown and the user is still listed as remnant:

```
#0 /var/www/nextcloud/apps/files_external/lib/MountConfig.php(100): OCA\User_LDAP\Handler\ExtStorageConfigHandler->handle()
#1 /var/www/nextcloud/apps/files_external/lib/Config/ConfigAdapter.php(76): OCA\Files_External\MountConfig::substitutePlaceholdersInConfig()
#2 /var/www/nextcloud/apps/files_external/lib/Config/ConfigAdapter.php(124): OCA\Files_External\Config\ConfigAdapter->prepareStorageConfig()
#3 [internal function]: OCA\Files_External\Config\ConfigAdapter->OCA\Files_External\Config\{closure}()
#4 /var/www/nextcloud/apps/files_external/lib/Config/ConfigAdapter.php(122): array_map()
#5 /var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php(124): OCA\Files_External\Config\ConfigAdapter->getMountsForUser()
#6 /var/www/nextcloud/lib/private/Files/SetupManager.php(226): OC\Files\Config\MountProviderCollection->addMountForUser()
#7 /var/www/nextcloud/lib/private/Files/SetupManager.php(318): OC\Files\SetupManager->OC\Files\{closure}()
#8 /var/www/nextcloud/lib/private/Files/SetupManager.php(227): OC\Files\SetupManager->setupForUserWith()
#9 /var/www/nextcloud/lib/private/Files/SetupManager.php(389): OC\Files\SetupManager->setupForUser()
#10 /var/www/nextcloud/lib/private/Files/Mount/Manager.php(95): OC\Files\SetupManager->setupForPath()
#11 /var/www/nextcloud/lib/private/Files/Filesystem.php(320): OC\Files\Mount\Manager->find()
#12 /var/www/nextcloud/lib/private/Files/View.php(1166): OC\Files\Filesystem::resolvePath()
#13 /var/www/nextcloud/lib/private/Files/View.php(378): OC\Files\View->basicOperation()
#14 /var/www/nextcloud/apps/onlyoffice/lib/fileversions.php(93): OC\Files\View->is_dir()
#15 /var/www/nextcloud/apps/onlyoffice/lib/fileversions.php(116): OCA\Onlyoffice\FileVersions::checkFolderExist()
#16 /var/www/nextcloud/apps/onlyoffice/lib/fileversions.php(305): OCA\Onlyoffice\FileVersions::getView()
#17 /var/www/nextcloud/apps/onlyoffice/lib/hooks.php(72): OCA\Onlyoffice\FileVersions::deleteAllVersions()
#18 /var/www/nextcloud/lib/private/legacy/OC_Hook.php(106): OCA\Onlyoffice\Hooks::userDelete()
#19 /var/www/nextcloud/lib/private/Server.php(588): OC_Hook::emit()
#20 [internal function]: OC\Server->OC\{closure}()
#21 /var/www/nextcloud/lib/private/Hooks/EmitterTrait.php(106): call_user_func_array()
#22 /var/www/nextcloud/lib/private/Hooks/PublicEmitter.php(40): OC\Hooks\BasicEmitter->emit()
#23 /var/www/nextcloud/lib/private/User/User.php(273): OC\Hooks\PublicEmitter->emit()
#24 /var/www/nextcloud/core/Command/User/Delete.php(65): OC\User\User->delete()
#25 /var/www/nextcloud/3rdparty/symfony/console/Command/Command.php(255): OC\Core\Command\User\Delete->execute()
#26 /var/www/nextcloud/core/Command/Base.php(177): Symfony\Component\Console\Command\Command->run()
#27 /var/www/nextcloud/3rdparty/symfony/console/Application.php(1009): OC\Core\Command\Base->run()
#28 /var/www/nextcloud/3rdparty/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#29 /var/www/nextcloud/3rdparty/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun()
#30 /var/www/nextcloud/lib/private/Console/Application.php(213): Symfony\Component\Console\Application->run()
#31 /var/www/nextcloud/console.php(100): OC\Console\Application->run()
#32 /var/www/nextcloud/occ(11): require_once('/var/www/nextcl...')
```

### Steps to reproduce

1. delete user from ldap
2. perform search of remants with ldap:show-remnants
3. try to delete the user by its nextcloudname with user:delete [nextcloudname]

### Expected behavior

The user should be finally deleted.

### Installation method

Nextcloud 25.03 running on Ubuntu 20.04 LTS

### Nextcloud Server version

25

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Nginx

### Database engine version

MariaDB

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - announcementcenter: 6.5.1
  - apporder: 0.15.0
  - bbb: 2.4.0
  - bruteforcesettings: 2.5.0
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - dav: 1.24.0
  - external: 5.0.2
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_accesscontrol: 1.15.1
  - files_antivirus: 4.0.3
  - files_external: 1.17.0
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - forms: 3.2.0
  - groupfolders: 13.1.1
  - keeweb: 0.6.12
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - onlyoffice: 7.8.0
  - password_policy: 1.15.0
  - passwords: 2023.3.11
  - previewgenerator: 5.2.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - related_resources: 1.0.3
  - richdocuments: 7.1.2
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_ldap: 1.15.0
  - viewer: 1.9.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit: 1.11.0
  - contacts: 5.2.0
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - encryption
  - extract: 1.3.5
  - files_markdown: 2.3.6
  - files_texteditor: 2.15.0
  - firstrunwizard: 2.10.0
  - integration_excalidraw: 2.0.1
  - photos: 1.3.0
  - recommendations: 1.0.0
  - sharebymail: 1.11.0
  - support: 1.4.0
  - survey_client: 1.9.0
  - suspicious_login
  - twofactor_totp
  - user_status: 1.1.1
  - weather_status: 1.1.0
```

Kind regards, Marco

**Comment by OneTwoBarbecue** [comment]
I updated to NC 25.0.5 but the problem still persists.

Kind regards, Marco

**Comment by joshtrichards** [comment]
Looks like it might be failing on some External Storage mounts. Does this user have any mounts?

`occ files_external:list`

**Comment by joshtrichards** [comment]
Your included output doesn't include the actual error, but I suspect this was the same as #34557 and is fixed in #37346.
